src/couch/src/couch_file.erl

```diff
@@ -906,6 +922,35 @@ reset_eof(#file{} = File) ->
     {ok, Eof} = file:position(File#file.fd, eof),
     File#file{eof = Eof}.
 
+write_checksum(Bin) when is_binary(Bin) ->
```

Comment:
I'd like to see an assertion that the result of this function is 16 bytes / 128 bits as this is assumed everywhere else, and as a guard against doing something silly in the future without realising.

```
-spec write_checksum(binary()) -> <<_:128>>.
write_checksum(Bin) when is_binary(Bin) ->
Result = case write_xxhash_checksums() of
    true -> exxhash:xxhash128(Bin);
    false -> couch_hash:md5_hash(Bin)
end,
<<_:128>> = Result.
```

that sort of thing
```

that sort of thing.

Reply:
and `generate_` instead of `write_` because this function generates the checksum but doesn't write it to disk.